controllers/datamovement_controller_test.go

```diff
@@ -21,35 +21,110 @@ package controllers
 
 import (
 	"context"
+	"fmt"
 	"os"
+	"path/filepath"
+	"time"
 
 	nnfv1alpha1 "github.com/NearNodeFlash/nnf-sos/api/v1alpha1"
+	"github.com/google/uuid"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	. "github.com/onsi/gomega/gstruct"
 
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// This script is used to generate output similar to `dcp`. It takes in a
+	// duration and interval that can be adjusted to meet the needs of the test.
+	// The duration is how long the fake data movement operation runs (in seconds)
+	// and the interval is the time between each progress output (using sleep).
+	//
+	// This is dumped into a temporary file and then ran as a bash script.
+	progressOutputScript = `#!/usr/bin/env bash
```

Comment:
consider using the `embed` functionality of go. That way this can be an actual `.sh` shell file with proper syntax highlighting, runnable, etc.

Reply:
Or just as a shell function. Should be callable from `make test`. You'd need to make sure you copy it into the Docker test image too.

Reply:
Thanks. I didn't know about embed. I'll go that route.